File: Java/BasicSample/app/src/main/java/com/neyogiry/android/sample/DateModel.java
package com.neyogiry.android.sample;

import android.os.Handler;

import java.util.ArrayList;

public class DateModel {

    ArrayList<String> mDates = new ArrayList<>();

    public DateModel() {
    }

    public void saveDate(final String date, final DateInterfaces.Model callback) {
        try{
            Handler handler = new Handler();
            handler.postDelayed(new Runnable() {
                @Override
                public void run() {
                    mDates.add(0, date);
                    onSuccess(callback);
                }
            }, 5000);
        }catch (Exception e){
            onError(callback, e.toString());
        }
    }

    public void onSuccess(DateInterfaces.Model callback) {
        callback.onResponse(mDates);
    }

    public void onError(DateInterfaces.Model callback, String error) {
        callback.onError(error);
    }

}

File: Kotlin/BasicSample/app/src/main/java/com/neyogiry/android/sample/DateModel.kt
package com.neyogiry.android.sample

import android.os.Handler

class DateModel {

    var mDates: ArrayList<String> = ArrayList()

    fun saveDate(date: String, callback: DateInterfaces.Model) {
        try {
            val handler = Handler()
            handler.postDelayed({
                mDates.add(0, date)
                onSuccess(callback)
            }, 5000)
        } catch (e: Exception) {
            onError(callback, e.toString())
        }

    }

    private fun onSuccess(callback: DateInterfaces.Model) {
        callback.onResponse(mDates)
    }

    private fun onError(callback: DateInterfaces.Model, error: String) {
        callback.onError(error)
    }

}